### ISSUE TITLE
Can I get same performance when I change pqos dynamically in one process?

### ISSUE THREAD

**Issue by skia92** [intel/intel-cmt-cat#93]
Hello,
Here is the problem what I am having.
Assume that there is a process and it has two functions: "A" function and "B" function I would say.
What I am trying to do is 

1. set COS 0 as 0xF and COS 1 as 0xF0, apply COS 0 on the process.
2. Do "A" function (It will spend the whole 4 way caches)
3. After "A" function finishes, change COS 0 to COS 1 on the same process.
4. start "B" function (I expect that it will spend another 4 ways caches)
5. Change COS 1 to COS 0 
6. Do "A" function again

In this situation, what I hope is that in step 6, there would be still data on the cache which are stored by "A" function. Thus, I expect that if I run "A" function again, there is no cache miss or very little. I've evaluated this process with MPKI metrics but I found quite high rate MPKI is estimated when I evaluted step 6, which means there is cache miss like cold miss. (I did use both interface: MSR & OS)

If step 6 MPKI should be the same or have little gap with the previous step 2, then probably I'm doing wrong so I will investigate my benchmark again. But, if it is a issue or it cannot be possible for some reason then please let me know. Any comment would be helped :+1:

**Comment by kmabbasi** [comment]
Hi Wonkyo,

Thanks for your feedback!

COS 0 is the default and it will be used by all processes on the system, so this data is likely to be evicted. You could rerun your benchmark on COS 1 and COS 2 and see if you get better performance.

Thanks,
Khawar

**Comment by skia92** [comment]
Hello Khawar,
I'm sorry but I think you couldn't catch my point.

My point was that can I still get cache hit in the default (e.g. COS 0) even if the process will run in different COS value in the meantime and come back to the default COS.

Anyway, when I opened this issue, I doubted my assumption but I was measuring wrong and I did get the right performance based on the assumption so I will close this issue.

Thanks,
